File: Assessment/src/pages/ShopPage.py
class ShopPage:

    def __init__(self, page):
        self.btn_frog = page.locator('//*[@id="product-2"]/div/p/a')
        self.btn_bunny = page.locator('//*[@id="product-4"]/div/p/a')
        self.btn_bear = page.locator('//*[@id="product-7"]/div/p/a')

        self.btn_cart = page.locator("#nav-cart > a")

        self.lbl_frog_price = page.locator('//*[@id="product-2"]/div/p/span')
        self.lbl_bunny_price = page.locator('//*[@id="product-4"]/div/p/span')
        self.lbl_bear_price = page.locator('//*[@id="product-7"]/div/p/span')

        self.lbl_frog = page.locator('//*[@id="product-2"]/div/h4')
        self.lbl_bunny = page.locator('//*[@id="product-4"]/div/h4')
        self.lbl_bear = page.locator('//*[@id="product-7"]/div/h4')

    # add stuffed frog to cart
    def click_frog(self, n):
        for i in range(n):
            self.btn_frog.click()

    # add fluffy bunny to cart
    def click_bunny(self, n):
        for i in range(n):
            self.btn_bunny.click()

    # add valentine bear to cart
    def click_bear(self, n):
        for i in range(n):
            self.btn_bear.click()

    # get stuffed frog price
    def get_frog_price(self):
        price = self.lbl_frog_price.inner_text().split(sep='$')
        return float(price[1])

    # get fluffy bunny price
    def get_bunny_price(self):
        price = self.lbl_bunny_price.inner_text().split(sep='$')
        return float(price[1])

    # get valentine bear price
    def get_bear_price(self):
        price = self.lbl_bear_price.inner_text().split(sep='$')
        return float(price[1])

    # get stuffed frog label
    def get_frog_label(self):
        return self.lbl_frog.inner_text()

    # get fluffy bunny label
    def get_bunny_label(self):
        return self.lbl_bunny.inner_text()

    # get valentine bear label
    def get_bear_label(self):
        return self.lbl_bear.inner_text()

    # function to click cart tab
    def click_cart(self):
        self.btn_cart.click()

File: Assessment/src/pages/ShopPage.ts
import { Page } from '@playwright/test'

export default class ShopPage { 
    constructor(private page: Page) {}

    btn_frog = '//*[@id="product-2"]/div/p/a'
    btn_bunny = '//*[@id="product-4"]/div/p/a'
    btn_bear = '//*[@id="product-7"]/div/p/a'
    btn_cart = '//*[@id="nav-cart"]/a'

    lbl_frog_price = '//*[@id="product-2"]/div/p/span'
    lbl_bunny_price = '//*[@id="product-4"]/div/p/span'
    lbl_bear_price = '//*[@id="product-7"]/div/p/span'

    lbl_frog = '//*[@id="product-2"]/div/h4'
    lbl_bunny = '//*[@id="product-4"]/div/h4'
    lbl_bear = '//*[@id="product-7"]/div/h4'

    // add stuffed frog to cart
    async click_frog(count: number) {
        for (let i = 0; i < count; i++) { 
            await this.page.click(this.btn_frog)
        }
    }

    // add fluffy bunny to cart
    async click_bunny(count: number) {
        for (let i = 0; i < count; i++) {
            await this.page.click(this.btn_bunny)
        }
    }

    // add valentine bear to cart
    async click_bear(count: number) {
        for (let i = 0; i < count; i++) {
            await this.page.click(this.btn_bear)
        }
    }

    // get stuffed frog price
    async get_frog_price() {
        let price = (await this.page.locator(this.lbl_frog_price).innerText()).toString().split('$')
        return Number(price[1])
    }

    // get fluffy bunny price
    async get_bunny_price() {
        let price = (await this.page.locator(this.lbl_bunny_price).innerText()).toString().split('$')
        return Number(price[1])
    }

    // get valentine bear price
    async get_bear_price() {
        let price = (await this.page.locator(this.lbl_bear_price).innerText()).toString().split('$')
        return Number(price[1])
    }

    // get stuffed frog label
    async get_frog_label() {
        return await this.page.locator(this.lbl_frog).innerText()
    }

    // get fluffy bunny label
    async get_bunny_label() {
        return await this.page.locator(this.lbl_bunny).innerText()
    }

    // get valentine bear label
    async get_bear_label() {
        return await this.page.locator(this.lbl_bear).innerText()
    }

    // function to click cart tab
    async click_cart() {
        await this.page.click(this.btn_cart)
    }
}